File: javascript/Always perfect.js
function checkRoot(string){

    const arr = string.split(',')
    const isNum = arr.every(s => s.trim() !== "" && !isNaN(s));
      
      if(arr.length !== 4) return 'incorrect input'
      if(!isNum) return 'incorrect input'
      
      for(let i =0; i <arr.length - 1; i++){
        if(Number(arr[i])+1 !== Number(arr[i + 1])){
        return 'not consecutive';
        }
      }
      
      let product = 1;
      for(let i =0; i < arr.length; i++){
        product *= Number(arr[i])
        
      }
      
      const result = product + 1
      const root = Math.sqrt(result)
       
       return `${result}, ${root}`
      }

File: python/Always perfect.py
import math

def check_root(strng):
    
    items = strng.split(',')
    
    if len(items) != 4:
        return 'incorrect input'
    
    nums = []
    
    for i in items:
        try:
            nums.append(int(i.strip()))
        except ValueError:
            return 'incorrect input'
        
    nums.sort()
    
    for i in range(3):
        if nums[i] + 1 != nums[i + 1]:
            return 'not consecutive'
    
    product = 1
    for i in nums:
        product *= i
            
    result = product + 1
    root = int(math.sqrt(result))
    
    return f'{result}, {root}'